docs/website/docs/developers/debugging/gpu-approaches-tips.md

```diff
@@ -0,0 +1,163 @@
+tractable.
+
+Once we have a more isolated case, the general methodology to pinpoint the
+exact culprit is to **1) collect and inspect the symptoms, 2) form hypothesis
+and run experiments to prove/refute the hypothesis, and 3) iterate**.
+
+### .. with shortcuts
+
+The above procedure is for facing a large problem with no clue, for example,
+when bringing up a new model end-to-end via IREE.
+
+Though most of the time, we can leverage existing facilities to avoid going down
+the full top-down hiearchical debugging procedure.
+For example, for regression happening on an existing model, CI or `git bitsect`
+might tell us directly the culprit commit.
+
+### .. using tools
+
+For issues with strong signals like crashing, it's also easier to pinpoint the
+exact culprit with dedicated tools--we can leverage various
+[sanitizers](./sanitizers.md) or debuggers.
+
+## Isolating the problematic component
+
+If we are facing a large problem without a clear clue, we need to isolate the
+problematic compiler or runtime layer first, typically by comparing with a
+working solution:
+
+* `[correctness/performance]` Sanitize the environment first--did you recently
+  updated the GPU SDK or driver?
+  Are others able to reproduce the issue?
+  If not what SDK / driver versions they are using?
+  Is your machine drawing enough power when benchmarking?
+  Is your machine connected with a mointor (for Vulkan)?
+  How long since you last rebooted your machine? 👻
+  Asking these questions and making sure the environment is proper can save
+  you hours of debugging sometimes.
+* `[correctness/performance]` We have multiple GPU targets/drivers in
+  IREE--LLVMGPU/CUDA, LLVMGPU/HIP, SPIR-V/Vulkan, SPIR-V/Metal.
+  For the _same_ GPU, we typically have two paths to target, e.g., CUDA/HIP
+  or Vulkan for NVIDIA/AMD GPUs, Metal or Vulkan for Apple GPUs.
+  If one path is correct/performant, we can diff against it to try isolate
+  the problem--the common/shared compiler/runtime code is likely okay; what
+  differs between paths is likely problematic.
+* `[correctness/performance][vulkan]` Vulkan supports different GPUs.
+  Similarly, if one GPU gives correct/performant result, we diff against it
+  to find clues.
+  Even more code in compiler/runtime are shared here; what's problematic is
+  likely different capabilities triggering different CodeGen pipelines so
+  revealing bugs in a particular CodeGen pipeline.
+  Or there are driver issues from a particular vendor.
+* `[correctness]` If the CPU is working properly, we can use the same dispatch
+  region formation and diff against the CPU dispatches one by one to isolate
+  the problem.
+* `[correctness]` `--iree-flow-trace-dispatch-tensors` and/or
+   `--iree-flow-break-dispatch=` to `iree-compile` is quite helpful to inspect
+   the output after all/each dispatch(es).
+* `[correctness]` `iree-reduce` is a great tool to reduce and isolate issues
+  programmatically.
```

Comment:
@Groverkss fyi, a mention of `iree-reduce` going in to the developer docs here. Is there a page somewhere that this could link to with more information? Mostly looking to add enough breadcrumbs for people reading this to know where to go next.

Could link to https://github.com/openxla/iree/pull/14892 and https://github.com/openxla/iree/blob/main/samples/reducer/README.md, for example.

Reply:
That should be fine for now. I can add a proper document page this week.

Reply:
Linked to the code. We can switch to point to the doc once it's up.